packages/dataviews/src/types.ts

```diff
@@ -141,4 +141,26 @@ export interface View {
 	 * The number of items per page
 	 */
 	perPage?: number;
+
+	/**
+	 * The hidden fields.
+	 */
+	hiddenFields: string[];
+}
+export interface ViewList extends ViewBase {
+	type: 'list';
+
+	layout: {
+		/**
+		 * The field to use as the primary field.
+		 */
+		primaryField: string;
+
+		/**
+		 * The field to use as the media field.
+		 */
+		mediaField: string;
+	};
 }
+
+export type View = ViewList | ViewBase;
```

Comment:
Ultimately, we need to find a way to make ViewBase generic (when we introduce view registration) but before going there this will probably become `ViewList | ViewTable | ViewGrid`

Reply:
Could you clarify what are we saying with this statement?

A concrete example: what happens if there's a view that uses `type: list` but doesn't declare the layout stuff as per the`ViewList`. Would it be considered a `ViewBase`?

Or perhaps I'm just confused because why any code would use `View`? Shouldn't it use `ViewList` (if it wants to enforce those things) or `ViewBase` (if it doesn't care)?

Reply:
> A concrete example: what happens if there's a view that uses type: list but doesn't declare the layout stuff as per theViewList. Would it be considered a ViewBase?

Right now it will be considered ViewBase because I made "layout" mandatory. But if "layout" is not mandatory (which I think it is), we should update the type and address the case where "layout" is undefined in the component itself.

> Or perhaps I'm just confused because why any code would use View? Shouldn't it use ViewList (if it wants to enforce those things) or ViewBase (if it doesn't care)?

It depends, the DataViews component should use the `View` type because it can render any View type but the `ViewList` component should only render the `ViewList` type and the typescript type checking will ensure that it's the case (when everything will be typed including the DataViews component itself). So assuming everything is typed, if I try to pass a view of type ViewGrid to the ViewList component, typescript will trigger an error.

Reply:
Looking more at our code, it seems we treat layout property as mandatory for ViewList, so I'm probably going to keep it as is.